lib/course_update_manager.rb

```diff
@@ -0,0 +1,66 @@
+#= Class for performing updates on data related to an individual Course
+class CourseUpdateManager
+  ################
+  # Entry points #
+  ################
+
+  def self.update_from_wiki(course, data={}, save=true)
+    require "#{Rails.root}/lib/importers/course_importer"
+    require "#{Rails.root}/lib/importers/user_importer"
+
+    id = course.id
+    if data.blank?
+      data = CourseImporter.get_course_info id
+      return if data.blank? || data[0].nil?
+      data = data[0]
+    end
+    # Symbol if coming from controller, string if from course importer
+    course.attributes = data[:course] || data['course']
+
+    return unless save
+    if data['participants']
```

Comment:
Can data['participants'] be `nil`? If it's an empty array, you don't need a check. (If it's not an empty array, it should be.)

```
[8] pry(main)> [].each_with_index do |x, i|
[8] pry(main)*   puts x
[8] pry(main)*   puts i
[8] pry(main)* end
```